airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/resumable_full_refresh_cursor.py

```diff
@@ -94,3 +95,15 @@ def get_request_body_json(
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Mapping[str, Any]:
         return {}
+
+
+@dataclass
+class SubstreamResumableFullRefreshCursor(ResumableFullRefreshCursor):
+    def close_slice(self, stream_slice: StreamSlice, *args: Any) -> None:
+        """
+        Once the current slice has finished syncing (paginator returns None),
+        we assume that the records are processed and emitted already,
+        thus we have to set the cursor to ` __ab_full_refresh_sync_complete: true `,
+        otherwise there is a risk of Inf. Loop processing the same slice.
+        """
+        self._cursor = FULL_REFRESH_COMPLETE_STATE
```

Comment:
this looks like closing any slice will result in setting the cursor to COMPLETE_STATE. Are we not closing individual slices?

Reply:
maybe we need to work on the naming of the class, but the reason why this works is that the `SubstreamResumableFullRefreshCursor` is only ever instantiated as a child partition cursor of a `PerPartitionCursor`. So when we call `close_slice()` we'll be closing that specific partition's slice. 

It does feel kind of weird that we can't completely reuse the regular `ResumableFullRefreshCursor`, but the closing behavior is slightly different than the normal one because we don't actually care about closing cursors with an actual value like a page number. We only care about closing it completely since it's invoked after we finish reading all pages.

Maybe we can call it `ChildPartitionResumableFullRefreshCursor`?

Reply:
got it! that makes sense. +1 for `ChildPartitionResumableFullRefreshCursor`! I think it's worth explaining somewhere. maybe in a docstring + RFR architecture doc since this isn't immediately obvious